polaris-react/src/utilities/merge.ts

```diff
@@ -24,9 +24,13 @@ export function merge<TSource1, TSource2, TSource3, TSource4, TSource5>(
   source5: TSource5,
 ): TSource1 & TSource2 & TSource3 & TSource4 & TSource5;
 export function merge<TResult>(...objs: any[]): TResult;
-export function merge<TSource1, TSource2, TSource3, TSource4, TSource5>(
-  ...objs: (TSource1 | TSource2 | TSource3 | TSource4 | TSource5)[]
-) {
+export function merge<
+  TSource1 extends GeneralObject,
+  TSource2 extends GeneralObject,
+  TSource3 extends GeneralObject,
+  TSource4 extends GeneralObject,
+  TSource5 extends GeneralObject,
+>(...objs: (TSource1 | TSource2 | TSource3 | TSource4 | TSource5)[]) {
```

Comment:
TypeScript wanted these types to `extend GeneralObject` so I just spammed it in here 🤷